include/pybind11/functional.h

```diff
@@ -50,9 +50,9 @@ struct type_caster<std::function<Return(Args...)>> {
         }
 
         auto src = reinterpret_borrow<object>(src_);
-        value = [src](Args... args) -> Return {
+        value = [src](Args&&... args) -> Return {
             gil_scoped_acquire acq;
-            object retval(src(std::move(args)...));
+            object retval(src(std::forward<Args>(args)...));
```

Comment:
Ah, cool.  Glad that there were unintended positive benefits.  The unit test for `MovableObject` has been added to the PR.